### PR TITLE
Fix #8 Scope JS output to Bootstrap transtypes

### LINKED ISSUE #8: Scope Lunr output to Bootstrap transtype
Just noticed while testing other output formats with this plugin installed that the `js/` folder is generated even when building unrelated output formats.

We should probably scope this to ensure the search index and related files are only generated when the output format is set to `html5-bootstrap`.

For one possible approach, see https://github.com/dita-ot/docs/pull/452#issuecomment-1491351762.

### PR DESCRIPTION
**Pull request by jason-fox**

### CODE REVIEW

File: plugin.xml [hunk 1]
```diff
@@ -5,12 +5,13 @@
...
 -->
 <plugin id="net.infotexture.dita-bootstrap.lunr" version="1.0.0">
-  <require plugin="net.infotexture.dita-bootstrap"/>
+  <require plugin="net.infotexture.dita-bootstrap" />
   <feature extension="ant.import" file="process_lunr.xml" />
   <!-- Ensure the plugin's error and logging messages are available -->
   <feature extension="dita.xsl.messages" file="resource/messages.xml" />
   <!-- HTML5 extension points -->   
   <feature extension="dita.xsl.html5" value="xsl/xhtml.xsl" type="file" />
+  <feature extension="depend.preprocess.pre" value="lunr.init" />
```
Comment (jason-fox) [inline plugin.xml:14]: Ideally this function should be switched to a `depend.ditabootstrap.pre` which is an extension point offered in the main plugin.

Comment (infotexture) [reply plugin.xml:14]: :+1 Makes sense.

**Review by infotexture — APPROVED** [review]
I verified that the changes in this branch indeed prevent the `js/` folder from appearing in default XHTML output, for example.

(Looks like a similar change might be necessary for keep `fox.jason.extend.css` from creating an empty `common-extended.css` file when generating output that doesn't require it.)

**Comment by infotexture** [comment]
> Ideally this function should be switched to a `depend.ditabootstrap.pre`

@jason-fox How would you prefer to handle this?

Shall we merge this as is, or would you prefer to add a new extension point to dita-bootstrap and use that here instead?

**Comment by jason-fox** [comment]
It would be better to create a new extension point. Please review  https://github.com/infotexture/dita-bootstrap/pull/91 first and then add this afterwards.

File: plugin.xml [hunk 2]
```diff
@@ -5,12 +5,13 @@
   See the accompanying LICENSE file for applicable licenses.
 -->
 <plugin id="net.infotexture.dita-bootstrap.lunr" version="1.0.0">
-  <require plugin="net.infotexture.dita-bootstrap"/>
+  <require plugin="net.infotexture.dita-bootstrap" />
```
Comment (infotexture) [inline plugin.xml:8]: This is consistent with the formatting of the surrounding lines, but in the main `dita-bootstrap` plug-in, we [enforce](https://github.com/infotexture/dita-bootstrap/blob/develop/.prettierrc.json#L15) the spaceless syntax with `"xmlSelfClosingSpace": false`, so we may want to align formatting here.

— But I can do that in a separate PR to keep things focused here, so fine for now.

File: process_lunr.xml [hunk 3]
```diff
@@ -13,20 +13,33 @@
...
+      </and>
     </condition>
+  </target>
 
+  <!--
+    Creates a JavaScript file for generating the lunr index
+  -->
+  <target
+    name="lunr.node.script.copy"
+    if="is.Bootstrap"
```
Comment (infotexture) [inline process_lunr.xml:40]: See previous comment

File: process_lunr.xml [hunk 4]
```diff
@@ -128,7 +147,12 @@
   <!--
     Check if Node.js is installed.
   -->
-  <target name="lunr.node.check">
+  <target
+    name="lunr.node.check"
+    if="is.Bootstrap"
```
Comment (infotexture) [inline process_lunr.xml:152]: ibid

**Review by infotexture — COMMENTED** [review]
Looks good, but a few questions…

File: process_lunr.xml [hunk 5]
```diff
@@ -13,20 +13,33 @@
...
   -->
-  <target name="lunr.init">
+  <target name="lunr.init" description="Setting up Lunr processing">
 
     <property
       name="lunr.lib"
       value="${dita.plugin.net.infotexture.dita-bootstrap.lunr.dir}/resource/lunr.js"
     />
 
+    <property name="is.Bootstrap" value="true" />
```
[the remark this answers is not in the document]
Comment (jason-fox) [reply process_lunr.xml:23]: That property is inside the `<target name="lunr.init">` and therefore only set if `lunr.init` is called - which is only done if `bootstrap.process.pre` is fired, which is the extension point.

The other targets are using the `depend.preprocess.copy-html.pre` extension point, so the `if` logic is only copying over the files if `bootstrap.process.pre` has been fired,

File: process_lunr.xml (hunk 3 above)
Comment (jason-fox) [reply process_lunr.xml:40]: Dependency of `lunr` target, it is only fired if `is.Bootstrap` has been set within the `lunr.init` target, which means `bootstrap.process.pre` has been fired.

File: process_lunr.xml (hunk 4 above)
Comment (jason-fox) [reply process_lunr.xml:152]: Dependency of `lunr` target, it is only fired if `is.Bootstrap` has been set within the `lunr.init` target, which means `bootstrap.process.pre` has been fired.

File: plugin.xml (hunk 2 above)
Comment (jason-fox) [reply plugin.xml:8]: Sounds like a separate PR, create a `.prettierrc.json` file and run it over everything.

Comment (infotexture) [reply plugin.xml:8]: Yep, will do.

File: process_lunr.xml (hunk 5 above)
[the remark this answers is not in the document]
Comment (infotexture) [reply process_lunr.xml:23]: OK, thanks for explaining. Sounds fine.

**Pull request merged**